modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotChecker.java

```diff
@@ -634,23 +634,16 @@ else if (!res.exceptions().isEmpty())
 
     /** */
     public static IdleVerifyResultV2 reduceHashesResults(
-        Map<ClusterNode, Map<PartitionKeyV2, PartitionHashRecordV2>> results,
+        Map<ClusterNode, Map<PartitionKeyV2, List<PartitionHashRecordV2>>> results,
         Map<ClusterNode, Exception> ex
     ) {
-        Map<PartitionKeyV2, List<PartitionHashRecordV2>> clusterHashes = new HashMap<>();
-
-        results.forEach((node, nodeHashes) -> {
-            assert ex.get(node) == null;
+        Map<PartitionKeyV2, List<PartitionHashRecordV2>> hashesRes = new HashMap<>();
 
-            for (Map.Entry<PartitionKeyV2, PartitionHashRecordV2> e : nodeHashes.entrySet()) {
-                List<PartitionHashRecordV2> records = clusterHashes.computeIfAbsent(e.getKey(), k -> new ArrayList<>());
-
-                records.add(e.getValue());
-            }
-        });
+        results.forEach((node, partsHashes) -> partsHashes.forEach((partKey, partHashLst) -> hashesRes.computeIfAbsent(partKey,
```

Comment:
Could you please replace it with for-loop? Three `->` in single construction is hard to read.